src/components/date/date.component.tsx

```diff
@@ -155,11 +156,19 @@ export const DateInput = React.forwardRef<HTMLInputElement, DateInputProps>(
     );
     const { inputRefMap, setInputRefMap } = useContext(DateRangeContext);
     const [open, setOpen] = useState(false);
-    const [selectedDays, setSelectedDays] = useState(
-      checkISOFormatAndLength(value)
+    const [selectedDays, setSelectedDays] = useState(() => {
+      const parsedDate = parse(value, "P", new Date(), {
```

Comment:
Comment: this potentially confusing as we already have a util called `parseDate` imported in the file. Maybe it would be worth using, or updating if needed, that function so we're consistent. We should also be able to use `isDateValid` from the same util file